File: Inventor-Training-Material-0c232f96f36e1e0bed31b07a6a6ebb901a31dd40/Module 10 - Events/Labs Solution - Events.C#/AddInEvents/StandardAddInServer.cs
// VBConversions Note: VB project level imports
using System.Collections;
using System;
using Microsoft.VisualBasic;
using System.Diagnostics;
// End of VB project level imports

using Inventor;
using System.Runtime.InteropServices;
using Microsoft.Win32;


namespace AddInEvents
{
	namespace AddInEvents
	{
		
		[ProgIdAttribute("AddInEvents.StandardAddInServer"), GuidAttribute("49d3dbd4-20c4-4c03-8027-49206f131596")]
        public class StandardAddInServer : Inventor.ApplicationAddInServer
		{
			
			// Inventor application object.
			private Inventor.Application m_inventorApplication;
			private Form1 mControlForm;
			
			private Inventor.ApplicationEvents m_ApplicationEvents;
			
			
#region ApplicationAddInServer Members
			
			
			public void Activate(Inventor.ApplicationAddInSite addInSiteObject, bool firstTime)
			{
				
				m_inventorApplication = addInSiteObject.Application;
				
				m_ApplicationEvents = m_inventorApplication.ApplicationEvents;
                m_ApplicationEvents.OnOpenDocument += this.m_ApplicationEvents_OnOpenDocument;
				
				mControlForm = new Form1(m_inventorApplication);
				mControlForm.ShowModeless();
				
			}
			
			private void m_ApplicationEvents_OnOpenDocument(Inventor._Document DocumentObject, string FullDocumentName, Inventor.EventTimingEnum BeforeOrAfter, Inventor.NameValueMap Context, Inventor.HandlingCodeEnum HandlingCode)
			{
				System.Windows.Forms.MessageBox.Show((string) ("OnOpenDocument: " + DocumentObject.DisplayName));
				
			}
			
			public void Deactivate()
			{
				
				// This method is called by Inventor when the AddIn is unloaded.
				// The AddIn will be unloaded either manually by the user or
				// when the Inventor session is terminated.
				
				// TODO:  Add ApplicationAddInServer.Deactivate implementation
				
				// Release objects.
				Marshal.ReleaseComObject(m_inventorApplication);
				m_inventorApplication = null;
				
				System.GC.WaitForPendingFinalizers();
				System.GC.Collect();
				
			}
			
			public object Automation
			{
				
				// This property is provided to allow the AddIn to expose an API
				// of its own to other programs. Typically, this  would be done by
				// implementing the AddIn's API interface in a class and returning
				// that class object through this property.
				
				get
				{
					return null;
				}
				
			}
			
			public void ExecuteCommand(int commandID)
			{
				
				// Note:this method is now obsolete, you should use the
				// ControlDefinition functionality for implementing commands.
				
			}
			
#endregion

            void m_ApplicationEvents_OnOpenDocument(_Document DocumentObject, string FullDocumentName, EventTimingEnum BeforeOrAfter, NameValueMap Context, out HandlingCodeEnum HandlingCode)
            {
                System.Windows.Forms.MessageBox.Show("OnOpenDocument fires!");

                HandlingCode = HandlingCodeEnum.kEventHandled;
            }
 
			
			
		}
		
	}
	
	
}

File: Inventor-Training-Material-0c232f96f36e1e0bed31b07a6a6ebb901a31dd40/Module 10 - Events/Labs Solution - Events.VB.Net/AddInEvents/StandardAddInServer.vb
Imports Inventor
Imports System.Runtime.InteropServices
Imports Microsoft.Win32

Namespace AddInEvents

    <ProgIdAttribute("AddInEvents.StandardAddInServer"), _
    GuidAttribute("49d3dbd4-20c4-4c03-8027-49206f131596")> _
    Public Class StandardAddInServer
        Implements Inventor.ApplicationAddInServer

        ' Inventor application object.
        Private m_inventorApplication As Inventor.Application
        Private mControlForm As Form1

        Private m_ApplicationEvents As Inventor.ApplicationEvents


#Region "ApplicationAddInServer Members"


        Public Sub Activate(ByVal addInSiteObject As Inventor.ApplicationAddInSite, ByVal firstTime As Boolean) Implements Inventor.ApplicationAddInServer.Activate

            m_inventorApplication = addInSiteObject.Application

            m_ApplicationEvents = m_inventorApplication.ApplicationEvents
            AddHandler m_ApplicationEvents.OnOpenDocument, AddressOf Me.m_ApplicationEvents_OnOpenDocument

            mControlForm = New Form1(m_inventorApplication)
            mControlForm.ShowModeless()

        End Sub

        Private Sub m_ApplicationEvents_OnOpenDocument(ByVal DocumentObject As Inventor._Document, _
                                                       ByVal FullDocumentName As String, _
                                                       ByVal BeforeOrAfter As Inventor.EventTimingEnum, _
                                                       ByVal Context As Inventor.NameValueMap, _
                                                       ByRef HandlingCode As Inventor.HandlingCodeEnum) _

            System.Windows.Forms.MessageBox.Show("OnOpenDocument: " + DocumentObject.DisplayName)

        End Sub

        Public Sub Deactivate() Implements Inventor.ApplicationAddInServer.Deactivate

            ' This method is called by Inventor when the AddIn is unloaded.
            ' The AddIn will be unloaded either manually by the user or
            ' when the Inventor session is terminated.

            ' TODO:  Add ApplicationAddInServer.Deactivate implementation

            ' Release objects.
            Marshal.ReleaseComObject(m_inventorApplication)
            m_inventorApplication = Nothing

            System.GC.WaitForPendingFinalizers()
            System.GC.Collect()

        End Sub

        Public ReadOnly Property Automation() As Object Implements Inventor.ApplicationAddInServer.Automation

            ' This property is provided to allow the AddIn to expose an API 
            ' of its own to other programs. Typically, this  would be done by
            ' implementing the AddIn's API interface in a class and returning 
            ' that class object through this property.

            Get
                Return Nothing
            End Get

        End Property

        Public Sub ExecuteCommand(ByVal commandID As Integer) Implements Inventor.ApplicationAddInServer.ExecuteCommand

            ' Note:this method is now obsolete, you should use the 
            ' ControlDefinition functionality for implementing commands.

        End Sub

#End Region
       
    End Class

End Namespace

